moderation: toggle comment date from timeago to full date

hjanuschka opened the pull request:
clicking on the "5 hours ago"  label toggles the full date and vic-a-verse

![bildschirmfoto 2018-09-03 um 09 05 06](https://user-images.githubusercontent.com/2891702/44972196-a3822b80-af58-11e8-89d2-2d7d478870e0.png)
![bildschirmfoto 2018-09-03 um 09 05 10](https://user-images.githubusercontent.com/2891702/44972197-a3822b80-af58-11e8-917a-bca658e66a7a.png)

hjanuschka committed:
toggle comment date

kgardnr commented:
Thank you @hjanuschka we will take a look!

@clintabrown if you and your team want to take a look too!

wyattjoh committed:
Merge branch 'master' into toggle_comment_date

alinhle commented on client/coral-admin/src/routes/Moderation/components/CommentTimeAgo.js:
```diff
@@ -0,0 +1,27 @@
+  }
+  render() {
+    return (
+      <div onClick={this.toggleDate.bind(this)}>
```
You can define _toggleDate_ as
```js
toggleDate = () => this.setState({ timeago: !this.state.timeago });
```

Then you don't need to bind this: `<div onClick={this.toggleDate}>`

alinhle commented on client/coral-admin/src/routes/Moderation/components/CommentTimeAgo.js:
```diff
@@ -0,0 +1,27 @@
+  }
+  render() {
+    return (
+      <div onClick={this.toggleDate.bind(this)}>
```
How about adding a CSS style with `cursor: pointer` indicating that it is clickable?

hjanuschka replied:
done

clintabrown commented on client/coral-admin/src/routes/Moderation/components/Comment.js:
```diff
@@ -127,7 +128,7 @@ class Comment extends React.Component {
               </span>
 
               <span className={styles.created}>
```
Would be nice if this element also became `<time className={styles.created} datetime={comment.created_at} title={comment.created_at}>` - more semantic and absolute time could also be viewed as a tooltip

hjanuschka replied:
done

clintabrown commented:
Could you also please apply this change to `client/coral-admin/src/components/UserDetailComment.js` too (moderation user profile)?

hjanuschka committed:
adress feedback

hjanuschka commented:
@clintabrown @cvle feedback addressed, i moved it to coral-framework.
and added the `title` thing, which is always the opposite of what you see.

also changed it in UserDetailComment.js and in the embed/stream too.


let me know if you'd need me to change anything else.
thx for taking the time to review this. ❤️

hjanuschka committed:
Merge branch 'toggle_comment_date' of github.com:hjanuschka/talk into toggle_comment_date

hjanuschka commented:
# Stream
![bildschirmfoto 2018-09-07 um 13 36 08](https://user-images.githubusercontent.com/2891702/45217225-94c2ae00-b2a3-11e8-887f-d325fc78393f.png)
![bildschirmfoto 2018-09-07 um 13 36 11](https://user-images.githubusercontent.com/2891702/45217226-955b4480-b2a3-11e8-96a9-f3184b00cb43.png)

# User Detail Comment

![bildschirmfoto 2018-09-07 um 13 36 29](https://user-images.githubusercontent.com/2891702/45217227-955b4480-b2a3-11e8-848d-cfcdcd4c0cad.png)
![bildschirmfoto 2018-09-07 um 13 36 34](https://user-images.githubusercontent.com/2891702/45217228-955b4480-b2a3-11e8-9316-c3b3b85f2cfc.png)

alinhle commented on client/coral-framework/components/TimeAgo.js:
```diff
@@ -0,0 +1,35 @@
+    return (
+      <span
+        onClick={this.toggleDate}
+        style={{ cursor: 'pointer' }}
```
We don't use `inline styles` because they can't be overwritten / customized using CSS. Can you add a CSS file and add this style using CSS modules?

hjanuschka replied:
done

alinhle commented on client/coral-framework/components/TimeAgo.js:
```diff
@@ -0,0 +1,35 @@
+}
+
+TimeAgo.propTypes = {
+  datetime: PropTypes.string,
```
Should also accept a `className` property which should be passed into `span`.

hjanuschka replied:
done

kgardnr committed:
Merge branch 'master' into toggle_comment_date

kgardnr commented:
@hjanuschka we've posted some feedback for your review!

cc @clintabrown

hjanuschka committed:
adress feedback

hjanuschka commented:
@kgardnr @cvle feedback addressed
